File: 0001-0100/0003-Longest Substring Without Repeating Characters/0003-Longest Substring  Without Repeating Characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashMap<Character, Integer> table = new HashMap();
        int start = 0, maxLen = 0;
        for (int i = 0; i < s.length(); ++i) {
            if (table.containsKey(s.charAt(i)) && table.get(s.charAt(i)) >= start) {
                start = table.get(s.charAt(i)) + 1;
            }
            else {
                maxLen = Math.max(maxLen, i - start + 1);
            }
            table.put(s.charAt(i), i);
        }

        return maxLen;
    }
}

File: 0001-0100/0003-Longest Substring Without Repeating Characters/0003-Longest Substring  Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        table = {}
        start = maxLen = 0
        for i, c in enumerate(s):
            if c in table and start <= table[c]:
                start = table[c] + 1
            else:
                maxLen = max(maxLen, i - start + 1)
            table[c] = i
        return maxLen
